etcdserver/api/membership/cluster.go

```diff
@@ -800,22 +807,43 @@ func ValidateClusterAndAssignIDs(lg *zap.Logger, local *RaftCluster, existing *R
 	return nil
 }
 
-func mustDetectDowngrade(lg *zap.Logger, cv *semver.Version) {
+func mustDetectDowngrade(lg *zap.Logger, cv *semver.Version, d *Downgrade) {
 	lv := semver.Must(semver.NewVersion(version.Version))
 	// only keep major.minor version for comparison against cluster version
 	lv = &semver.Version{Major: lv.Major, Minor: lv.Minor}
-	if cv != nil && lv.LessThan(*cv) {
-		if IsVersionChangable(cv, lv) {
+
+	if d.Enabled && !d.TargetVersion.Equal(*lv) {
+		if lg != nil {
+			lg.Fatal(
+				"invalid downgrade; server version does not match the downgrade target version",
+				zap.String("current-server-version", version.Version),
+				zap.String("target-cluster-version", d.TargetVersion.String()),
+			)
+		} else {
+			plog.Fatalf("invalid downgrade; server version does not match the downgrade target version "+
+				"(current version: %s, target cluster version: %s).", version.Version, d.TargetVersion.String())
+		}
+		return
+	}
+
+	if cv != nil {
+		if d.Enabled && d.TargetVersion.Equal(*lv) {
 			if lg != nil {
 				lg.Info(
 					"cluster is downgrading to current server version",
 					zap.String("current-server-version", version.Version),
 					zap.String("determined-cluster-version", version.Cluster(cv.String())),
+					zap.String("target-cluster-version", d.TargetVersion.String()),
 				)
 			} else {
-				plog.Infof("cluster is downgrading to current version: %s from determined cluster version: %s).", version.Version, version.Cluster(cv.String()))
+				plog.Infof("cluster is downgrading to target version: %s, "+
+					"determined cluster version: %s, current-server-version: %s).",
+					d.TargetVersion.String(), version.Cluster(cv.String()), version.Version)
```

Comment:
I know d.TargetVersion.Equal(*lv) now, but better keep lg and plog output the same string? 
lg: cluster is downgrading to current server version
plog: cluster is downgrading to target version
And the order is different.